dev-util/kbuild/files/kbuild-0.1.9998.3499-int-conversion.patch

```diff
@@ -0,0 +1,26 @@
+Fix int to pointer conversion for clang
+
+Note - this parameter was not actually used inside eval_buffer because of a define.
+
+Bug: https://bugs.gentoo.org/875449
+
+--- a/src/kmk/loadapi.c
++++ b/src/kmk/loadapi.c
+@@ -20,6 +20,8 @@
+ #include "variable.h"
+ #include "dep.h"
+ 
++#include <string.h>
++
+ /* Allocate a buffer in our context, so we can free it.  */
+ char *
+ gmk_alloc (unsigned int len)
+@@ -59,7 +61,7 @@
+   install_variable_buffer (&pbuf, &plen);
+ 
+   s = xstrdup (buffer);
+-  eval_buffer (s, flp IF_WITH_VALUE_LENGTH_PARAM (strlen (s) /** @todo suboptimal */));
++  eval_buffer (s, flp IF_WITH_VALUE_LENGTH_PARAM (strchr (s, 0) /** @todo suboptimal */));
```

Comment:
Bit confused -- why couldn't we just then pass s?

Reply:
That is supposed to be the end of the string, i.e., `s + strlen(s)`.  We could pass `s` or even `NULL` and it would work (inside the function that parameter is only used inside an `ifdef` that is not defined), but this is the correct pointer it would expect if it used it.

This is how the function looks like (I simplified it a bit):
```
void
eval_buffer (char *buffer, const floc *flocp, char *eos)
{
  struct ebuffer ebuf;

#ifndef CONFIG_WITH_VALUE_LENGTH
  ebuf.size = strlen (buffer);
#else
  ebuf.size = eos - buffer;
  ebuf.eol = eos;
  assert(strchr(buffer, '\0') == eos);
#endif
```